pma_api/api_1_0/datalab.py

```diff
@@ -35,9 +35,13 @@ def get_datalab_combos():
     elif not indicator and char_grp:
         json_obj = DatalabData.combos_char_grp(char_grp)
         return jsonify(json_obj)
-    # else: survey_list, indicator, char_grp are all None
-    # TODO (jkp 2017-08-29) put in informative error code. Need: more time
-    return 'Request args are all empty', 400
+    elif survey_list is None and indicator is None and char_grp is None:
```

Comment:
Doesn't the linter dislike this `elif`? This final block of code should not be part of an `elif` or `else`. I agree with the JSON error message style that you have.

Reply:
The advantage of the `elif` is explicitness. By not having an `else` or a trailing return at the end of the function, we're declaring a finite set of specific cases we expect. If none of those cases are met, the function will not return, which should cause an exception since it's a route function. I think that's the behavior we want.

I don't know if I checked the linter when I made the change, but I checked it just now and it is not complaining.

Reply:
I disagree. This final block is not an `elif`. The condition that you have written is the logical equivalent to `else`. Either it should be under `else` or it should not be in the if-elif-else at all as I had originally. All outcomes should be accounted for, and they should all return something.

Reply:
Hey James. I just looked at the code block and you are correct that the `elif` case here is the logical equivalent of an `else` or just having the code block after the conditionals.

This doesn't throw lint warnings or anything, so the only difference here is stylistic; the terminal `else` case simply states very explicitly its conditions without the reader of the code having to look at the rest of the conditional. I would say that in some cases this is a good thing to do stylistically. That is, as a conditional block increases in complexity, a terminal case with explicit conditions increases in usefulness. If so, the inverse would hold that as a conditional block increases in simplicity, an explicit terminal case is less useful.

I don't have strong feelings about the stylistic use here in this particular case, but following my own explanation it would seem that it is concurrent with your call here in that the conditional isn't particularly complex or hard to follow, so an explicit terminal case is not especially useful.

I'll remove and let you know when the changes are pushed.